### DISCUSSION TITLE
How ResizeWithPadOrCropd worked?

### DISCUSSION THREAD

**Discussion by talhaanwarch** [Project-MONAI/MONAI#2590] [Q&A]
First of all thanks for such an amazing library and for the quick responses. 
Second, I have some questions related to ResizeWithPadOrCropd.
Does it pad at the end or the start of the image (one side)? or all sides of the images?
When It cropped, does it cropped from all sides? Is it possible to crop that area that doesn't have any foreground? I mean give more importance to area which has foreground.
Thanks

**Comment by Nic-Ma** [comment]
Hi @talhaanwarch ,

Thanks for your interest and investigation.
I think we need to enhance the API to support different padding methods, I will try to enhance it ASAP.

Thanks.

**Reply by Nic-Ma** [reply]
Hi @talhaanwarch ,

I submitted PR to enhance the pad args: https://github.com/Project-MONAI/MONAI/pull/2594.
For the cropping, we use `CenterCrop` in ResizeWithPadOrCropd, if you want to use other advanced crop transform, I think you can use some `pad` + `crop` 2 transforms.

Thanks.
